### PR TITLE
Change phone number validations

### PR DESCRIPTION
**Pull request by sanchita**

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenControllerTest.kt [hunk 1]
```diff
@@ -41,9 +45,9 @@ class RegistrationPhoneScreenControllerTest {
   }
 
   @Test
-  fun `when screen is created and a local user is present, then the local user must be cleared`() {
+  fun `when screen is created and existing ongoing entry is present, then the local user must be cleared`() {
```
Comment (sanchita) [inline app/src/test/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenControllerTest.kt]: @vinaysshenoy Can you please review this test? It seems like the test was... wrong? But it still passed?

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryScreenController.kt [hunk 2]
```diff
@@ -237,12 +243,13 @@ class PatientEntryScreenController @Inject constructor(
...
 
-    val phoneNumberErrors: Observable<List<PatientEntryValidationError>> = events
-        .ofType<PatientEntrySaveClicked>()
+    val phoneNumberErrors = events.ofType<PatientEntrySaveClicked>()
         .withLatestFrom(phoneNumberChanges, noPhoneToggles)
         .map { (_, phoneNumber, isNoneSelected) ->
           when {
             (phoneNumber.isBlank() && !isNoneSelected) -> listOf(PHONE_NUMBER_EMPTY)
+            (numberValidator.validate(phoneNumber, LANDLINE_OR_MOBILE) == LENGTH_LESS) -> listOf(PHONE_NUMBER_LENGTH_LESS)
+            (numberValidator.validate(phoneNumber, LANDLINE_OR_MOBILE) == LENGTH_MORE) -> listOf(PHONE_NUMBER_LENGTH_MORE)
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/newentry/PatientEntryScreenController.kt]: TODO: discuss in person

File: app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
-  fun isValid(number: String): Boolean
+  enum class Type {
+    LANDLINE_OR_MOBILE,
+    MOBILE
+  }
+
+  enum class Result {
+    VALID,
+    LENGTH_LESS,
+    LENGTH_MORE
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/registration/phone/PhoneNumberValidator.kt]: TODO: discuss in person

File: app/src/main/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenController.kt [hunk 4]
```diff
@@ -82,11 +84,10 @@ class RegistrationPhoneScreenController @Inject constructor(
...
         .map { it.phoneNumber }
 
-    return events
-        .ofType<RegistrationPhoneDoneClicked>()
+    return events.ofType<RegistrationPhoneDoneClicked>()
         .withLatestFrom(phoneNumberTextChanges)
-        .map { (_, number) -> numberValidator.isValid(number) }
-        .filter { isValidNumber -> isValidNumber.not() }
+        .map { (_, number) -> numberValidator.validate(number, MOBILE) }
+        .filter { it != VALID }
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenController.kt]: TODO: discuss in person

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -66,6 +66,8 @@
   <string name="patiententry_error_both_dateofbirth_and_age_empty">Enter patient’s age</string>
   <string name="patiententry_error_invalid_dateofbirth">Enter patient’s date of birth in DD/MM/YYYY</string>
   <string name="patiententry_error_dateofbirth_is_in_future">Fix patient’s date of birth</string>
+  <string name="patiententry_error_phonenumber_length_less">Phone number cannot be less than 6 digits</string>
+  <string name="patiententry_error_phonenumber_length_more">Phone number cannot be more than 12 digits</string>
```
Comment (saket) [inline app/src/main/res/values/strings.xml]: TODO: discuss in person

**Pull request merged**